include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -536,15 +536,15 @@ class CBlitImageFilter :
 								if (!srcPix[0])
 									continue;
 
-								auto sample = lineBuffer+i*MaxChannels;
-								value_t swizzledSample[MaxChannels];
+								// TODO: make sure there is no leak due to ChannelCount!
```

Comment:
@achalpandeyy what's with the TODO?

Reply:
your `onDecode` uses `decodePixelsRuntime` which needs a place to  store all 4 channels, so `preSwizzleSample` needs to be MaxChannels or 4 elements in size

Reply:
the swizzle probably as well, basically unless you make `onDecode` take a `channelCount` and copy stuff over, you need `MaxChannelCount` on all arrays

Actually thinking of this, its best you address the TODO here:
https://github.com/Devsh-Graphics-Programming/Nabla/blob/fb457e9d39fede6d125ab75e3e823dfd4ab2f3bc/include/nbl/asset/filters/CSwizzleableAndDitherableFilterBase.h#L356

declare a `Tdec tmp[4];` to store the swizzle results to, then copy `channelCount` values of `tmp` into `decodeBuffer`

Reply:
resolved in following commit, we are doing it in place now like @achalpandeyy suggested in his TODO

Reply:
and it will break because swizzles permute RGBA channels around and you need a auxiliary scratch array!

Reply:
ok I see the temp has moved to `onDecode`, but issues persist